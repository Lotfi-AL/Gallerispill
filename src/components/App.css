body {
    margin: auto;
    font-family: "Press Start 2P", cursive;
}
