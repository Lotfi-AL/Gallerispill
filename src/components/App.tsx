import React, { Component, useState } from "react";
import Landingpage from "./Landingpage/Landingpage";
import Poetry from "./Poetry/Poetry";
import Game from "./Game/Game";
import { StatusContext, statusType } from "./Store/StatusContext";
import Audio from "./ControlPanel/Audio";
import { StatusProvider } from "./Store/StatusProvider";
import Ask from "./Store/StatusContextExample";
import Terrain from "./Terrain/Terrain";
import ControlPanel from "./ControlPanel/ControlPanel";

type MyState = {
    landed: boolean;
};

interface ToggleProps {
    ClickHandler: () => void;
}
export type saveState = {
    landed: boolean;
};

export class App extends React.Component<{}, MyState> {
    state: MyState = { landed: window.sessionStorage.getItem("landed") === "true" };

    constructor(props: any) {
        super(props);
        this.toggleClickHandler = this.toggleClickHandler.bind(this);
    }

    toggleClickHandler = () => {
        console.log("click");
        this.setState((prevState) => {
            if (prevState.landed === true) {
                return { landed: false };
            }
            if (prevState.landed === false) {
                return { landed: true };
            }
        });
    };

    render() {
        if (!this.state.landed) {
            return (
                <StatusProvider>
                    <Landingpage ClickHandler={this.toggleClickHandler}></Landingpage>
                </StatusProvider>
            );
        }
        return (
            <StatusProvider>
                <Terrain></Terrain>
                <ControlPanel></ControlPanel>
                <Game></Game>

                <Poetry></Poetry>
            </StatusProvider>
        );
    }
}
