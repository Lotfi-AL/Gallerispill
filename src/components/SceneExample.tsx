import React from "react";
import { useStatus } from "./Store/StatusProvider";
export default function SceneExample() {
    const { scene, setScene, currScene, setCurrScene } = useStatus();

    const leggo = () => {
        console.log("hey");
        console.log(scene);
        const currStatus = scene[currScene];
        console.log(currStatus);
    };
    return (
        <div>
            <button onClick={leggo}> Trykk p√• meg</button>
        </div>
    );
}
