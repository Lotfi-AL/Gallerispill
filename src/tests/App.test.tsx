import { App } from "../components/App";
import renderer from "react-test-renderer";
import React from "react";

test("app_snapshot", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
