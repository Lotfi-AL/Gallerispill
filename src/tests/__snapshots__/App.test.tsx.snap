// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app_snapshot 1`] = `
<div
  className="landingpage"
>
  <div
    className="instructions box"
  >
    <h1>
      Description
    </h1>
    <p>
      This is a 2d immersive poem experience. Move with arrow keys and interact with scene components using the spacebar. Read along with the poems as they are highlighted and go through the doors to switch poem and scene. As you go along you can change the scene to your liking by interacting with the paintings.
    </p>
  </div>
  <div
    className="credits"
  >
    <div
      className="container box"
    >
      <p
        className="text"
      >
        Created by
      </p>
      <ul
        className="list"
      >
        <li
          className="item"
        >
          Yeti
        </li>
        <li
          className="item"
        >
          Hobbit
        </li>
        <li
          className="item"
        >
          Gorilla
        </li>
      </ul>
    </div>
  </div>
  <div
    className="box link startLink"
    onClick={[Function]}
  >
    Load Previous Save
  </div>
  <div
    className="box link startLink"
    onClick={[Function]}
  >
    Click to start!
  </div>
  <div
    className="box"
  >
    <h2>
      Assets and sources
    </h2>
    <ul>
      <li>
        <a
          href="https://poetrydb.org/index.html"
          target="_blank"
        >
          Poetry
        </a>
      </li>
      <li>
        SFX
        <ul>
          <li>
            <a
              href="https://www.fesliyanstudios.com/royalty-free-sound-effects-download/rain-21"
              target="_blank"
            >
              Rain
            </a>
          </li>
          <li>
            <a
              href="https://www.fesliyanstudios.com/royalty-free-sound-effects-download/wind-blowing-37"
              target="_blank"
            >
              Wind
            </a>
          </li>
        </ul>
      </li>
      <li>
        <a
          href="https://oddpotatogift.itch.io/16x16-fantasy-pack"
          target="_blank"
        >
          House tileMap
        </a>
      </li>
      <li>
        <a
          href="https://thkaspar.itch.io/micro-character-bases"
          target="_blank"
        >
          Character
        </a>
      </li>
      <li>
        <a
          href="https://vnitti.itch.io/grassy-mountains-parallax-background"
          target="_blank"
        >
          Terrain
        </a>
      </li>
      <li>
        <a
          href="https://fonts.google.com/specimen/Press+Start+2P?sidebar.open=true&selection.family=Press+Start+2P"
          target="_blank"
        >
          Fonts
        </a>
      </li>
      <li>
        <a
          href="https://codepen.io/yoannhel/pen/sJpDj"
          target="_blank"
        >
          Credits animation
        </a>
      </li>
    </ul>
  </div>
</div>
`;
